File: src/main/java/edu/recepespire/service/IngredientService.java
package edu.recepespire.service;

import edu.recepespire.dto.IngredientDTO;
import edu.recepespire.entity.Ingredient;

import java.util.List;

public interface IngredientService {
    Ingredient createIngredient(IngredientDTO dto);
    List<Ingredient> getAllIngredients();
    Ingredient getIngredientById(Long id);
    Ingredient updateIngredient(Long id, IngredientDTO dto);
    void deleteIngredient(Long id);
}

File: recipesphere-frontend/src/app/services/ingredient.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Ingredient, IngredientDTO } from '../models/ingredient.model';

@Injectable({
  providedIn: 'root'
})
export class IngredientService {
  private apiUrl = 'http://localhost:8080/api/ingredients';

  constructor(private http: HttpClient) {}

  getAllIngredients(): Observable<Ingredient[]> {
    return this.http.get<Ingredient[]>(this.apiUrl);
  }

  getIngredientById(id: number): Observable<Ingredient> {
    return this.http.get<Ingredient>(`${this.apiUrl}/${id}`);
  }

  createIngredient(ingredient: IngredientDTO): Observable<Ingredient> {
    return this.http.post<Ingredient>(`${this.apiUrl}/single`, ingredient);
  }

  createIngredients(ingredients: IngredientDTO[]): Observable<Ingredient[]> {
    return this.http.post<Ingredient[]>(`${this.apiUrl}/bulk`, ingredients);
  }

  updateIngredient(id: number, ingredient: IngredientDTO): Observable<Ingredient> {
    return this.http.put<Ingredient>(`${this.apiUrl}/${id}`, ingredient);
  }

  deleteIngredient(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}